kernel/src/engine/arrow_expression.rs

```diff
@@ -30,6 +32,18 @@ use crate::schema::{DataType, PrimitiveType, SchemaRef};
 use crate::{EngineData, ExpressionEvaluator, ExpressionHandler};
 
 // TODO leverage scalars / Datum
+//
+pub fn expression_to_row_filter(predicate: Expression) -> RowFilter {
+    let arrow_predicate = ArrowPredicateFn::new(ProjectionMask::all(), move |batch| {
```

Comment:
Okay so we'd definitely like projection. I was initially only thinking about the row group skipping optimization. 

Re option 2: I could imagine a query `select * from table where value > 50`. Here I believe the filter would only have the `value` column, but we are looking for all columns. If that's the case, I think option 1 is the way.

Reply:
Couple of things:
1. For actual "data path" we don't push the predicate down in the "all in one" scan (see [here](https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/scan/mod.rs#L285)). Although an engine could choose to do that since `read_parquet_files` does take a predicate. Filtering like this is not always a win for reading parquet, so we'll need to be a bit more careful about when we want to actually push things down. Our [example](https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/examples/read-table-multi-threaded/src/main.rs#L294) code also doesn't push it down yet. Once this is working, we can maybe think about providing some guidance to engines about when to push down.
2. my option 1 vs 2 above was just about figuring out _which_ columns to project out to evaluate the predicate. You _definitely_ want to project columns out. So for  your example, you could either require the caller to tell you that it's expression requires the `value` column, or you could look at the expression itself and notice that that's the only column it references. Figuring it out from the expression is much nicer for users, but requires more work because you need need to examine the expression, see which cols it references. Regardless you then have to figure out what the positions of those columns are in the parquet file, which is very non-trivial (see `get_requested_indices`). You might want to limit this to only allow filtering on root cols.

Reply:
Update on this: I'm now filtering columns, but only at the root level since `StructType::project` only works at root level. I extract the columns by recursing down the expression structure. I've put up an issue to explore projections in nested columns #353.